### PR TITLE
New Identity system updates

### PR DESCRIPTION
**Pull request by medikoo**
_In theory they're breaking changes, but as we do not use that API anywhere yet (and we can call it experimental at this point) it's safe to push it with a minor release_

- Replace `auth.isAuthenticated` with `auth.resolveAuthenticationMode` which also provides an answer to how we're logged in (whether as a user, or whether as an org)
- Rename `auth.resolveIdToken` into `auth.resolveToken` as it can return either an _id token_ or _org token_
- Remove `auth.getOrgId`, as it's not really an _auth_ method, and the approach is replaced with a more generic `apiRequest`
- Introduce `apiRequest` a generic request handler for any new identity system API handling

### CODE REVIEW

File: auth/resolve-authentication-mode.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+'use strict';
+
+const resolveAuthToken = require('./resolve-token');
+
+const isOrgAuthentication = Boolean(process.env.SLS_ORG_TOKEN);
```
Comment (pgrzesik) [inline auth/resolve-authentication-mode.js:5]: Is this token specific to Console only or it also works for Dashboard?

File: api-request.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
+'use strict';
+
+const ensureString = require('type/string/ensure');
+const fetch = require('node-fetch');
+const log = require('./log').log.get('auth');
+const ServerlessError = require('./serverless-error');
+const backendUrl = require('./lib/auth/urls').backend;
+const resolveAuthToken = require('./auth/resolve-token');
+
+module.exports = async (pathname) => {
```
Comment (pgrzesik) [inline api-request.js:10]: This util is only for new identity system, correct?

**Review by pgrzesik — DISMISSED** [review]
Looks good in general 👍 I have a few clarification questions

File: auth/resolve-authentication-mode.js (hunk 1 above)
Comment (medikoo) [reply auth/resolve-authentication-mode.js:5]: It's specific to new identity system (so not applicable for Dashbaord)

File: api-request.js (hunk 2 above)
Comment (medikoo) [reply api-request.js:10]: Yes

**Comment by pgrzesik** [comment]
Thanks for clarifications, looks good to me 👍

**Comment by medikoo** [comment]
> I see that we add new API in this package, but which other CLI uses this package?

Currently no other CLI, but it's hosted here as any other CLI which would want to publish to Console would have to use same endpoints.

**Pull request merged**